### PR TITLE
Optimization: magento/module-catalog is_null change to strict comparison

### PR DESCRIPTION
**Pull request by coderimus**
Hello,
This is the one of the optimization PR linked with the `is_null()` change. I decided to do them per module just to avoid any mistake during delivery and review.
Best regards,
Alex

### Description
**Micro-optimizations for Magento\Catalog**
Magento2 has next sniff [MicroOptimizations](https://github.com/magento/magento2/blob/2.2-develop/dev/tests/static/framework/Magento/Sniffs/MicroOptimizations/IsNullSniff.php) It checks next: `is_null must be avoided. Use strict comparison instead.`
Some classes are under `// @codingStandardsIgnoreFile` such as Magento\Catalog module models, block and cannot be checked during static tests. In this PR I replaced `is_null` with strict comparison only for models and block in Magento\Catalog module.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios
N/A

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Block/Adminhtml/Category/Tree.php [hunk 1]
```diff
@@ -228,7 +228,7 @@ public function getStoreSwitcherHtml()
     public function getLoadTreeUrl($expanded = null)
     {
         $params = ['_current' => true, 'id' => null, 'store' => null];
-        if (is_null($expanded) && $this->_backendSession->getIsTreeWasExpanded() || $expanded == true) {
+        if (($expanded === null) && $this->_backendSession->getIsTreeWasExpanded() || $expanded == true) {
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Block/Adminhtml/Category/Tree.php]: Would be better to remove brackets near ```$expanded === null```. They're not needed there

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php [hunk 2]
```diff
@@ -313,9 +313,9 @@ public function getOptionValues()
                     $value['checkboxScopeTitle'] = $this->getCheckboxScopeHtml(
                         $option->getOptionId(),
                         'title',
-                        is_null($option->getStoreTitle())
+                        $option->getStoreTitle() === null
                     );
-                    $value['scopeTitleDisabled'] = is_null($option->getStoreTitle()) ? 'disabled' : null;
+                    $value['scopeTitleDisabled'] = ($option->getStoreTitle() === null) ? 'disabled' : null;
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php]: There also

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php [hunk 3]
```diff
@@ -379,9 +379,9 @@ public function getOptionValues()
                         $value['checkboxScopePrice'] = $this->getCheckboxScopeHtml(
                             $option->getOptionId(),
                             'price',
-                            is_null($option->getStorePrice())
+                            $option->getStorePrice() === null
                         );
-                        $value['scopePriceDisabled'] = is_null($option->getStorePrice()) ? 'disabled' : null;
+                        $value['scopePriceDisabled'] = ($option->getStorePrice() === null) ? 'disabled' : null;
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Edit/Tab/Options/Option.php]: And there

File: app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php [hunk 4]
```diff
@@ -1092,7 +1092,7 @@ public function getSelectCountSql()
     protected function _getSelectCountSql($select = null, $resetLeftJoins = true)
     {
         $this->_renderFilters();
-        $countSelect = is_null($select) ? $this->_getClearSelect() : $this->_buildClearSelect($select);
+        $countSelect = ($select === null) ? $this->_getClearSelect() : $this->_buildClearSelect($select);
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php]: And there

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by coderimus** [comment]
@ihor-sviziev thank you for your review! I made requested changes according to your review. Please, check. :)

File: app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php [hunk 5]
```diff
@@ -1466,17 +1466,17 @@ public function addPriceData($customerGroupId = null, $websiteId = null)
     {
         $this->_productLimitationFilters->setUsePriceIndex(true);
 
-        if (!isset($this->_productLimitationFilters['customer_group_id']) && is_null($customerGroupId)) {
+        if (!isset($this->_productLimitationFilters['customer_group_id']) && ($customerGroupId === null)) {
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php]: Please remove unnecessary brackets.

File: app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php [hunk 6]
```diff
@@ -1466,17 +1466,17 @@ public function addPriceData($customerGroupId = null, $websiteId = null)
...
             $customerGroupId = $this->_customerSession->getCustomerGroupId();
         }
-        if (!isset($this->_productLimitationFilters['website_id']) && is_null($websiteId)) {
+        if (!isset($this->_productLimitationFilters['website_id']) && ($websiteId === null)) {
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/ResourceModel/Product/Collection.php]: And there

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@Coderimus few more issues are left. Sorry for not sending it at first time

**Comment by coderimus** [comment]
@ihor-sviziev thank you for your comments and detail review! I added needed changes, please, check.

**Review by ihor-sviziev — APPROVED** [review]

**Pull request merged**